nmigen/lib/roundrobin.py

```diff
@@ -0,0 +1,47 @@
+from .. import *
+
+
+__all__ = ["RoundRobin"]
+
+
+class RoundRobin(Elaboratable):
+    """Round-robin scheduler.
+
+    For a given set of requests, the round-robin scheduler will
+    grant one request.
+
+    Parameters
+    ----------
+    width : int
+        Number of requests.
+
+    Attributes
+    ----------
+    requests : Signal(width), in
+        Set of requests.
+    grant : Signal(range(width)), out
+        Number of the granted request.
```

Comment:
What happens if no request is active?

Reply:
grant isn't updated and keeps its value. Should I add that to the doc?

Reply:
I think it would be better to take a page from `lib.coding` and make an explicit `valid` output. (Actually, `lib.coding` uses a weird `n` convention for that--I think `valid` would be far better.)

Reply:
#dbd3926 adds a valid signal.